File: problem_7/solution.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

// Where m represents total ways of steps at a time
func stepsCount(n int, m int) int {
	arr := make([]int, n+1)
	arr[0] = 1
	arr[1] = 1
	for i := 2; i <= n; i++ {
		arr[i] = 0
		j := 1
		for j <= m && j <= i {
			arr[i] += arr[i-j]
			j++
		}
	}
	return arr[n]
}

func main() {
	file, err := os.Open("test_cases.txt")
	if err != nil {
		fmt.Println(err)
	}

	defer file.Close()

	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		content := strings.Split(scanner.Text(), " ")
		steps, err := strconv.Atoi(content[0])
		if err != nil {
			fmt.Println(err)
		}
		ans, err := strconv.Atoi(content[1])
		if err != nil {
			fmt.Println(err)
		}
		if stepsCount(steps, 2) == ans {
			fmt.Println("Passed")
		} else {
			fmt.Println("Failed")
		}
	}
}

File: problem_7/solution.py
# Where m represents total ways of steps at a time
def steps_count(n, m=2):
    arr = [0 for a in range(n+1)]
    arr[0] = arr[1] = 1
    for i in range(2, n+1):
        j = 1
        while j <= m and j <= i:
            arr[i] += arr[i - j]
            j += 1
    return arr[n]


if __name__ == '__main__':
    filename = "test_cases.txt"
    with open(filename) as file:
        for line in file:
            steps, ans = line.split()
            steps = int(steps)
            ans = int(ans)

            if steps_count(steps) == ans :
                print("Passed")
            else:
                print("Failed")
